### ISSUE TITLE
Functionality to easily copy output of last calculation to clipboard

### ISSUE THREAD

**Issue by hamirmahal** [sharkdp/numbat#394]

**Comment by hamirmahal** [comment]
I think it'd be useful if there was some sort of keybinding to automatically copy the result of the last calculation to the keyboard. This is a bit more convenient for users who don't want to switch over to precisely highlighting text with a mouse.

**Comment by hamirmahal** [comment]
https://askubuntu.com/questions/759358/select-copy-paste-in-terminal-using-only-the-keyboard/759487#759487 has an interesting option, `screen`, but so far it seems like that just lets me copy and paste in the terminal, whereas I'd like to be able to copy text from the terminal, and then paste it into an editor, which `screen` doesn't seem to do.

If it's not too much work, maybe `numbat` could implement this functionality itself.

**Comment by rusinikita** [comment]
I started using numbat for system design calculation. I think it's useful to have ability to copy in markdown to paste in notion, coda.io, etc.

**Comment by rusinikita** [comment]
Do you know the codebase? I think we can do fork or PR

**Comment by rben01** [comment]
Once PR #555 lands I can work on adding a `copy` command. I need to think about 1. how to recognize passed options, and what options should be passable (`--commas`, `--group-char=' '`, `--scientific`, `--no-units`, ...) I don't really want to implement general CLI parsing here, but output formatting is one place that options abound 2. how to configure default options (env variable? config file?)

**Comment by sharkdp** [comment]
> 1. how to recognize passed options, and what options should be passable (`--commas`, `--group-char=' '`, `--scientific`, `--no-units`, ...) I don't really want to implement general CLI parsing here, but output formatting is one place that options abound

Hm, I really don't think commands should have options. I don't think anyone is going to type something like `copy --group-char=' ' --scientific`. It's faster to just type `copy` and then edit the pasted quantity as needed.

Adding configuration options in Numbat's config file is something we can certainly add if needed, but I would first try to start with a good default, and see how far it takes us.

How that default format for copying should look like is an open question. It's probably *not* the same format that we use for displaying. For example, the underscore-decimal separators are used in some other languages and tools (e.g. Python), but it's far from being a standard. I would probably not add decimal separators at all in the default format.

> 2. how to configure default options (env variable? config file?)

Just like we do with other config options. Via Numbats config file (https://numbat.dev/doc/cli-customization.html#configuration). And possibly via command-line options, if we think this is useful.

**Comment by rben01** [comment]
Makes sense. Does Numbat have any clipboard functionality currently, or should I add a crate for it in order to implement it? If so, which one? https://github.com/1Password/arboard looks good to me.

**Comment by sharkdp** [comment]
> should I add a crate for it in order to implement it

Yes, please do. I don't have experience with any clipboard library, so please feel free to make a decision.

Maybe it's worth thinking about the wasm version as well. Clipboard functionality is very much platform specific, but we should still strive to share code. At least the "what exactly should be put into the clipboard" part. But maybe even more. Maybe there is a lightweight cross-platform clipboard that also supports browsers.
